apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoader.java

```diff
@@ -294,23 +311,18 @@ record -> {
                 "%s failed to complete processing the records in time: '%s'.",
                 this.getClass().getSimpleName(), dataToLoad));
     } catch (InterruptedException e) {
-      // Interrupts should not be used on this thread, so go boom.
-      throw new RuntimeException(e);
+      LOGGER.error("Encountered an unexpected InterruptedException, load failed.");
+      error.compareAndSet(null, e);
     }
 
     /*
-     * If any batch load tripped a fatal error, we should throw an exception
-     * on this thread to kill the pipeline. Exceptions thrown from the batch thread with the
-     * error are ignored/lost, so have this synchronized boolean signal to do this specifically on the main thread.
-     * We also can't use a passed-in error handler from up the execution chain or else the pipeline manager shutdown procedure
-     * waits for this thread to shut down, which is waiting on the above 72-hour timeout before it unblocks.
-     *
-     * FUTURE: If we wish to be more discerning about recoverable errors, we could return false here which will
-     * allow the pipeline to continue operating if an error is considered non-fatal (it also currently moves the failed files
-     * to "Done" instead of "Failed", which would need to be fixed.)
+     * If any batch load tripped a fatal error, we call the error handler.
+     * The stream will have already cleanly halted processing.
```

Comment:
If part of the error handler process is waiting for all processing threads to terminate, as it did in the past, it'll sit there waiting for the full timeout above on awaitTermination (which is set to multiple days to allow the loads to run in a long-long scenario). Can't discern by looking at the new flow if that will be a problem. When I was testing this scenario to do the bugfix that added fatalFailure, I had two test cases I (manually) ran to test the "hung pipeline" scenario; one where a rif file had an invalid data item in a row (say, a number that was too big to fit in int) and one where the header column name was bad (messing up the name of a header.) 

In retrospect I probably should have written an actual test to cover that case, but that'd be something it'd be good to make sure can't happen so we don't revive https://jira.cms.gov/browse/BFD-2329

Reply:
Good question.  Previously the batch processing was stream was not tied into the error handling logic so it only terminated when it hit end of file.

This change in `RifLoader` (and same right below it for alternative stream) stops parsing/queuing batches once an exception has been reported:

```
            .takeWhile(record -> error.get() == null) // stop if an exception is thrown
```

Likewise this change in `DefaultDataSetMonitorListener` does the same at the manifest level.

Reply:
Thanks! Seems the new flow will take care of this. I meant to mention above, but we can probably remove fatalFailure boolean then, I think it's been obsoleted.